tb/cocotb/soc/test_timer.py

```
@@ -0,0 +1,282 @@
+#   0x0C  TMR_PRESCALE RW [15:0]  0 → tick every clock
+#
+# Timing notes:
+#   - TMR_COUNTER is updated every clock; AXI reads have latency of a few
+#     cycles, so counter assertions use "strictly increasing" or bounded-delta
+#     tolerances rather than exact equality.
+#   - Each .write() / .read() consumes multiple clock cycles; the counter
+#     advances during transactions.  All tolerance windows account for this.
+
+import cocotb
+from cocotb.clock import Clock
+from cocotb.triggers import RisingEdge, ClockCycles
```

Flake8 (7.3.0)

[error] 'cocotb.triggers.ClockCycles' imported but unused

(F401)